### ISSUE TITLE
Cannot read property 'match' of undefined

### ISSUE THREAD

**Issue by jackjackde** [SnowdogApps/magento2-frontools#349]
clean frontools install
nodejs: 10.15.0
npm: 6.5.0
frontools 1.6.0

on `npm install` installation breaks after "gulp-util@3.0.8"
log: error Cannot read property 'match' of undefined
from log file:

```
verbose stack TypeError: Cannot read property 'match' of undefined
1517 verbose stack     at tarballToVersion (/usr/lib/node_modules/npm/lib/install/inflate-shrinkwrap.js:87:20)
1517 verbose stack     at inflatableChild (/usr/lib/node_modules/npm/lib/install/inflate-shrinkwrap.js:99:22)
1517 verbose stack     at BB.each (/usr/lib/node_modules/npm/lib/install/inflate-shrinkwrap.js:55:12)
1517 verbose stack     at tryCatcher (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
1517 verbose stack     at Object.gotValue (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/reduce.js:155:18)
1517 verbose stack     at Object.gotAccum (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/reduce.js:144:25)
1517 verbose stack     at Object.tryCatcher (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
1517 verbose stack     at Promise._settlePromiseFromHandler (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:512:31)
1517 verbose stack     at Promise._settlePromise (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:569:18)
1517 verbose stack     at Promise._settlePromise0 (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:614:10)
1517 verbose stack     at Promise._settlePromises (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:694:18)
1517 verbose stack     at _drainQueueStep (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:138:12)
1517 verbose stack     at _drainQueue (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:131:9)
1517 verbose stack     at Async._drainQueues (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:147:5)
1517 verbose stack     at Immediate.Async.drainQueues [as _onImmediate] (/usr/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:17:14)
1517 verbose stack     at runCallback (timers.js:705:18)
```

i found this issue: [https://github.com/npm/npm/issues/20954](https://github.com/npm/npm/issues/20954)

**Comment by Igloczek** [comment]
Please use Yarn instead of NPM until I manage to fix this problem.

**Comment by jackjackde** [comment]
i do and the installation runs fine. But now i have another problem and it's difficult to describe...

If i start a session from browser for the first time (before i run gulp dev or gulp styles) everything is fine, every external file is loading. After 'gulp dev | styles' magento load the "regular" page but not the js | img files from pub directory.

**Comment by Igloczek** [comment]
Without logs from console and browser, I cannot say more than "sorry for inconveniences" 😉 

But if you have problems even with images, which are not processed by the Frontools at all, it's probably a problem with your M2 instance configuration.

**Comment by mageho** [comment]
Good afternoon,

If it could help us to have a clue on what's go'in on :   
`

1517 verbose stack TypeError: Cannot read property 'match' of undefined
1517 verbose stack     at tarballToVersion (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/lib/install/inflate-shrinkwrap.js:87:20)
1517 verbose stack     at inflatableChild (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/lib/install/inflate-shrinkwrap.js:99:22)
1517 verbose stack     at BB.each (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/lib/install/inflate-shrinkwrap.js:55:12)
1517 verbose stack     at tryCatcher (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
1517 verbose stack     at Object.gotValue (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/reduce.js:155:18)
1517 verbose stack     at Object.gotAccum (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/reduce.js:144:25)
1517 verbose stack     at Object.tryCatcher (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
1517 verbose stack     at Promise._settlePromiseFromHandler (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:512:31)
1517 verbose stack     at Promise._settlePromise (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:569:18)
1517 verbose stack     at Promise._settlePromise0 (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:614:10)
1517 verbose stack     at Promise._settlePromises (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:693:18)
1517 verbose stack     at Async._drainQueue (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:133:16)
1517 verbose stack     at Async._drainQueues (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:143:10)
1517 verbose stack     at Immediate.Async.drainQueues [as _onImmediate] (/home/artbambou/.nvm/versions/node/v10.15.0/lib/node_modules/npm/node_modules/bluebird/js/release/async.js:17:14)
1517 verbose stack     at runCallback (timers.js:705:18)
1517 verbose stack     at tryOnImmediate (timers.js:676:5)
1518 verbose cwd /home/artbambou/public_html/new/vendor/snowdog/frontools
1519 verbose Linux 4.9.58-xxxx-std-ipv6-64
1520 verbose argv "/home/artbambou/.nvm/versions/node/v10.15.0/bin/node" "/home/artbambou/.nvm/versions/node/v10.15.0/bin/npm" "install"
1521 verbose node v10.15.0
1522 verbose npm  v6.4.1
1523 error Cannot read property 'match' of undefined
1524 verbose exit [ 1, true ]

`

Cheers,
Ilan PARMENTIER

**Comment by Igloczek** [comment]
@mageho Thanks for the report, but it's exactly the same as OP show in the first comment, so I removed yours.

**Comment by nghelo** [comment]
delete npm-shrinkwrap.json and try again with `npm install`

**Comment by dasiux** [comment]
> delete npm-shrinkwrap.json and try again with `npm install`

this worked fine so far, no need for yarn.

**Comment by henkvalk** [comment]
> delete npm-shrinkwrap.json and try again with `npm install`

Thanks for the fix. This was solving my issue

**Comment by Igloczek** [comment]
Fixed in v1.7.0

**Comment by vishwanath231** [comment]
thanks for the fix.
